client/db/src/lib.rs

```diff
@@ -1707,12 +1709,16 @@ impl<Block: BlockT> Backend<Block> {
 		finalized: NumberFor<Block>,
 		displaced: &FinalizationOutcome<Block::Hash, NumberFor<Block>>,
 	) -> ClientResult<()> {
-		if let BlocksPruning::Some(blocks_pruning) = self.blocks_pruning {
-			// Always keep the last finalized block
-			let keep = std::cmp::max(blocks_pruning, 1);
-			if finalized >= keep.into() {
-				let number = finalized.saturating_sub(keep.into());
-				self.prune_block(transaction, BlockId::<Block>::number(number))?;
+		if BlocksPruning::AllWithNonFinalized == self.blocks_pruning {
```

Comment:
Let's replace this if with `match` so that this code requires explicit update if new options are added to  `BlocksPruning` in the future